### DISCUSSION TITLE
Recommended courses for learning ML?

### DISCUSSION THREAD

**Discussion by deliquentspacexer** [MarsArtistsCommunity/CommunityOverview#7] [Q&A] [answered]
Hi everybody,

I've been reading about Physics-Informed Neural Networks (PINN) from several sources, and I would appreciate it if anyone would direct me to online courses that would help a physics graduate to get started with Machine Learning? 

Thank you!

**Comment by oceanbluesky** [comment, marked as answer]
There's this remarkable video series: https://www.youtube.com/playlist?list=PLMrJAkhIeNNQ0BaKuBKY43k4xMo6NSbBa

And of course several well-regarded university courses have been archived online: 

Intros to various course offerings:
https://ai.stanford.edu/stanford-ai-courses/ 

Physics and ML
https://youtu.be/fiX8c-4K0-Q?si=VDp2R5Hjq-ERoq8e 

CS106A, B, and X
https://youtube.com/playlist?list=PLoCMsyE1cvdWiqgyzwAz_uGLSHsuYZlMX&si=Usd8W_PCV9hp3OZ5
https://online.stanford.edu/courses/cs106b-programming-abstractions 

CS109 Intro to Probability for Computer Scientists
https://www.youtube.com/playlist?list=PLoROMvodv4rOpr_A7B9SriE_iZmkanvUg 

Coursera Google:
https://www.coursera.org/learn/gcp-fundamentals/home/module/1

AWS:
https://skillbuilder.aws/

**Comment by deliquentspacexer** [comment]
Thank you! 

Are you studying CS now?

**Comment by oceanbluesky** [comment]
Pacing it out. Probably shifting into formal studies of Physics rather than CS, given AI's capabilities. 

How did you like the open house?

**Comment by deliquentspacexer** [comment]
Eh. Room full of identical annoying minds. Would have enjoyed it more if conversations could be had over the music and demos or activities were scheduled. I'm concerned it could be like that at the hackathon. Really unplanned, poorly organized.

**Comment by oceanbluesky** [comment]
Here are some other courses online you may find worthwhile:

YOUTUBE LECTURES:
Karpathy: https://www.youtube.com/results?search_query=karpathy+CVPR+2020 
CS106B: https://www.youtube.com/watch?v=Ua-31ucGAZ0&t=3s (2018)
CS107: https://www.youtube.com/watch?v=Ps8jOj7diA0&t=23s (2010 but good instructor)
CS109: https://www.youtube.com/watch?v=2MuDZIAzBMY&list=PLoROMvodv4rOpr_A7B9SriE_iZmkanvUg (2023)
CS236: https://www.youtube.com/watch?v=XZ0PMRWXBEU&t=23s (4 months ago) <---

**Comment by oceanbluesky** [comment]
And some of the prerequisites for formal classes at Stanford:

REREQUISITES FOR 236:
CS 221 Artificial Intelligence
CS 229 Machine Learning
CS 228 Probabilistic Graphical Models
CS230 Deep Learning

PREREQUISITES FOR 229: 
CS106B: Programming Abstractions
https://online.stanford.edu/courses/cs106b-programming-abstractions
CS109: Intro to Probability for Computer Scientists
https://online.stanford.edu/courses/cs109-introduction-probability-computer-scientists 

STATS116: Theory of Probability
https://online.stanford.edu/courses/stats116-theory-probability 
MATH51: Linear Algebra, Multivariable Calculus, and Modern Applications
https://online.stanford.edu/courses/math51-linear-algebra-multivariable-calculus-and-modern-applications 

PREREQUISITES FOR 221:
Programming
CS103 Discrete Mathematics
CS109 Probability for Computer Scientists
Math 51 Linear Algebra and Multivariable Calculus
Further reading: There are no required textbooks for this class, and you should be able to learn everything from the lecture notes and homeworks. However, if you would like to pursue more advanced topics or get another perspective on the same material, here are some great resources:
Russell and Norvig. Artificial Intelligence: A Modern Approach. A comprehensive reference for all the AI topics that we will cover. http://aima.cs.berkeley.edu/
Koller and Friedman. Probabilistic Graphical Models. Covers factor graphs and Bayesian networks (this is the textbook for CS228). http://mitpress.mit.edu/books/probabilistic-graphical-models
Sutton and Barto. Reinforcement Learning: An Introduction. Covers Markov decision processes and reinforcement learning (free online). http://incompleteideas.net/book/the-book-2nd.html
Hastie, Tibshirani, and Friedman. The Elements of Statistical Learning. Covers machine learning from a rigorous statistical perspective (free online). https://web.stanford.edu/~hastie/ElemStatLearn/
Tsang. Foundations of Constraint Satisfaction. Covers constraint satisfaction problems (free online). http://www.bracil.net/edward/FCS.html

**Comment by deliquentspacexer** [comment]
Wow! Thank you so much!!

**Comment by oceanbluesky** [comment]
Sure, good luck with your studies

**Comment by PhdPIML** [comment]
Hey all I thought that was a good party!
